app/controllers/users_controller.rb

```diff
@@ -0,0 +1,59 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def login_form
+    @user = User.new
+  end
+
+  def login
+    @user = User.find_by(name: params[:user][:name])
+
+    if @user.nil?
+      @user = User.new(name: params[:user][:name])
+      if !@user.save
```

Comment:
Style: I'd recommend using `unless` here for clarity:

```suggestion
      unless @user.save
```